File: (8 kyu) Counting sheep/(8 kyu) Counting sheep.rs
// #1
// fn count_sheep(sheep: &[bool]) -> u8 {
//     sheep.into_iter().filter(|&&x| x).collect::<Vec<_>>().len() as u8
// }
// #2
fn count_sheep(sheep: &[bool]) -> u8 {
    sheep.iter().filter(|&&x| x).count() as u8
}

File: 8kyu/(8 kyu) Counting sheep/(8 kyu) Counting sheep.cpp
#include <vector>
using namespace std;

// #1
// int count_sheep(vector<bool> arr)
// {
//     int res = 0;
//     for (auto i : arr)
//     {
//         if (i)
//             res++;
//     }
//     return res;
// }

// #2
#include <algorithm>
int count_sheep(std::vector<bool> arr)
{
    return std::count(arr.begin(), arr.end(), true);
}